### PR TITLE
Add POC for RUTv2 with JS v9 modular SDK.

### PR DESCRIPTION
**Pull request by yuchenshi**
This adds some minimal tests (assertsFail and assertSucceed, authed and unauthed, RTDB and Firestore) to show the format of the new Rules Unit Testing library (RUT). This PR also functions as a summary of the changes necessary to start testing in an existing project.

Happy testing! 

🌹🍒🍓🌶🎒🎈🍎 🧪🍀 💚 🥝 🍵 🌵🍏

### CODE REVIEW

**Review by rachelmyers — APPROVED** [review]
This looks great! :shipit:

**Review by markarndt — APPROVED** [review]

**Comment by rachelmyers** [comment]
The failing tests in [firestore.spec.js](https://github.com/firebase/quickstart-testing/blob/master/unit-test-security-rules/test/firestore.spec.js) and [database.spec.js](https://github.com/firebase/quickstart-testing/blob/master/unit-test-security-rules/test/database.spec.js) are blocking the update to v9. I'll make sure that any tests there are recreated in the new tests, and then replace those files with these.

File: unit-test-security-rules-v9/test/database.spec.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+let aliceDb;
+/** @type testing.RulesTestContext */
+let bobDb;
+
+before(async () => {
+  testEnv = await initializeTestEnvironment({
+    database: {rules: readFileSync('database.rules.json', 'utf8')},
+  });
+
+  unauthedDb = testEnv.unauthenticatedContext().database();
```
Comment (yuchenshi) [inline unit-test-security-rules-v9/test/database.spec.js:38]: nit: I'd suggest inlining these to make each snippet more standalone and copypasteable. ([Tests too DRY? Make them DAMP!](https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html)) Creating an instance per test also reduces chances of cache-related issues.

File: unit-test-security-rules-v9/test.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { initializeTestEnvironment, assertFails, assertSucceeds } from '../../firebase-js-sdk/packages/rules-unit-testing';
```
Comment (yuchenshi) [inline unit-test-security-rules-v9/test.ts:1]: Delete this

File: unit-test-security-rules-v9/test/firestore.spec.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+before(async () => {
+  // Silence expected rules rejections from Firestore SDK. Unexpected rejections
+  // will still bubble up and will be thrown as an error (failing the tests).
+  setLogLevel('error');
+
+  testEnv = await initializeTestEnvironment({
+    firestore: {rules: readFileSync('firestore.rules', 'utf8')},
+  });
+
+  unauthedDb = testEnv.unauthenticatedContext().firestore();
```
Comment (yuchenshi) [inline unit-test-security-rules-v9/test/firestore.spec.js:40]: ditto

File: unit-test-security-rules-v9/test.ts (hunk 2 above)
Comment (yuchenshi) [reply unit-test-security-rules-v9/test.ts]: Whoops the whole file is a mistake. I can take care of this

**Comment by yuchenshi** [comment]
Merged, but note that tests for v9 directory won't pass until https://github.com/firebase/firebase-js-sdk/pull/5501/files is fixed. I'll enable CI for the directory after that.

**Pull request merged**